source/MapPanel.cpp

```diff
@@ -789,6 +791,10 @@ void MapPanel::Select(const System *system)
 		if(isJumping)
 			plan.push_back(source);
 	}
+
+	// Reset the travel destination if the final system in the travel plan has changed.
+	if(!plan.empty() && plan.front() != previousFinalSystem)
```

Comment:
If there was a previous plan, then the new destination system is guaranteed to be different (because the function returns early if that's the case). If there was no previous plan, and we arrive here then the destination is the player system. So the second condition can be dropped.

```suggestion
	if(!plan.empty())
```